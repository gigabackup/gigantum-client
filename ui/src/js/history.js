import { createBrowserHistory } from 'history';

/**
*  @param {}
*  Mehtod gets basename for history object
*  @return {String}
*/
const getBasename = () => {
  const { pathname } = document.location;
  const pathList = pathname.split('/');
  const uniqueClientString = (pathList.length > 2)
    ? pathList[2]
    : '';
  const basename = (process.env.BUILD_TYPE === 'cloud')
    ? `/run/${uniqueClientString}/`
    : '/';
  return basename;
};

const basename = getBasename();

export default createBrowserHistory({
  basename,
  forceRefresh: false,
});
