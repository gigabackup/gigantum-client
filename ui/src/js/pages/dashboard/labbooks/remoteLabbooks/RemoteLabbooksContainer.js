// vendor
import {
  createFragmentContainer,
  graphql,
} from 'react-relay';
// components
import Labbooks from '../Labbooks';

export default createFragmentContainer(
  Labbooks,
  {
    labbookList: graphql`
      fragment RemoteLabbooksContainer_labbookList on LabbookQuery{
        labbookList{
          id
          ...RemoteLabbooks_remoteLabbooks
        }
      }
    `,
  },
);
