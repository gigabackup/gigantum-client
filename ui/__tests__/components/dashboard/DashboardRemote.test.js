
import React from 'react';
import renderer from 'react-test-renderer';
import history from 'JS/history';
import { mount } from 'enzyme';
import RemoteLabbooksContainer from 'Components/dashboard/labbooks/remoteLabbooks/RemoteLabbooksContainer';
import { Provider } from 'react-redux';
import store from 'JS/redux/store';
import relayTestingUtils from '@gigantum/relay-testing-utils';
import json from './__relaydata__/DashboardRemote.json';


history.location.pathname = 'hostname/labbooks/cloud';

const fixtures = {
  auth: () => {},
  labbookList: json.data,
  remoteLabbooks: json.data.labbookList,
  history: {
    replace: () => jest.fn(),
    location: {
      hash: '',
    },
  },
  refetchSort: () => {},
  loading: false,
};

describe('DashboardRemote', () => {
    it('Renders a snapshot', () => {
      const wrapper = renderer.create(
         relayTestingUtils.relayWrap(
          <Provider store={store}>
            <RemoteLabbooksContainer {...fixtures} />
          </Provider>,
          {}, json.data,
        ),
      );

      const tree = wrapper.toJSON();

      expect(tree).toMatchSnapshot();
    });
});
