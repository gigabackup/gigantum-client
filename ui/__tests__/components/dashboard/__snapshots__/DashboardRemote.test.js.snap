// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRemote Renders a snapshot 1`] = `
<div
  className="Labbooks"
>
  <div />
  <div
    className="Labbooks__panel-bar"
  >
    <h6
      className="Labbooks__username"
    >
      uitest
    </h6>
    <h2
      className="Labbooks__title"
    >
      Projects
    </h2>
  </div>
  <div
    className="Labbooks__menu  mui-container flex-0-0-auto"
  >
    <ul
      className="Labbooks__nav  flex flex--row"
    >
      <li
        className="Labbooks__nav-item Labbooks__nav-item--local"
      >
        <a
          onClick={[Function]}
        >
          Local
        </a>
      </li>
      <li
        className="Labbooks__nav-item Labbooks__nav-item--cloud Labbooks__nav-item--selected"
      >
        <a
          onClick={[Function]}
        >
          Cloud
        </a>
      </li>
      <hr
        className="Labbooks__navigation-slider Labbooks__navigation-slider--cloud"
      />
      <div
        className="hidden cloudLocal"
      >
        <div
          className="ToolTip__toggle cloudLocal"
          onClick={[Function]}
        >
          <div
            className="ToolTip__glow-container"
          >
            <div
              className="ToolTip__glow-ring-outer"
            >
              <div
                className="ToolTip__glow-ring-inner"
              />
            </div>
          </div>
        </div>
        <div
          className="hidden cloudLocal"
        />
        <div
          className="hidden cloudLocal"
        >
          This allows you to switch between Local and Cloud Project listings. Both Projects you have created and Projects that have been shared with you will be visible.
        </div>
      </div>
    </ul>
  </div>
  <div
    className="Labbooks__subheader grid"
  >
    <div
      className="Labbooks__search-container column-2-span-6 padding--0"
    >
      <input
        className="Labbooks__search no--margin"
        defaultValue=""
        onFocus={[Function]}
        onKeyUp={[Function]}
        placeholder="Filter Projects by name or description"
        type="text"
      />
    </div>
    <div
      className="LabbookFilterBy column-4-span-3 padding--0"
    >
      Filter by:
      <span
        className="LabbookFilterBy__selector Dropdown Dropdown--collapsed"
        onClick={[Function]}
      >
        All
      </span>
      <ul
        className="LabbookFilterBy__menu Dropdown__menu box-shadow hidden"
      >
        <li
          className="LabbookFilterBy__list-item"
          onClick={[Function]}
        >
          All 
          ✓ 
        </li>
        <li
          className="LabbookFilterBy__list-item"
          onClick={[Function]}
        >
          My Projects 
          
        </li>
        <li
          className="LabbookFilterBy__list-item"
          onClick={[Function]}
        >
          Shared with me 
          
        </li>
      </ul>
    </div>
    <div
      className="LabbookSort column-4-span-3 padding--0"
    >
      Sort by:
      <span
        className="LabbookSort__selector Dropdown Dropdown--collapsed"
        onClick={[Function]}
      >
        Modified Date (Newest)
      </span>
      <ul
        className="LabbookSort__menu Dropdown__menu box-shadow hidden"
      >
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Modified Date (Newest) 
          ✓ 
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Modified Date (Oldest) 
          
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Creation Date (Newest) 
          
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Creation Date (Oldest) 
          
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          A-Z 
          
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Z-A 
          
        </li>
      </ul>
    </div>
  </div>
  <div
    className="Labbooks__listing"
  >
    <div
      className="grid"
    >
      <div
        className="Card Card--300 column-4-span-3 flex flex--column justify--space-between"
      >
        <div
          className="RemoteLabbooks__row RemoteLabbooks__row--icon"
        >
          <button
            className="RemoteLabbooks__icon RemoteLabbooks__icon--cloud Tooltip-data Tooltip-data--small"
            data-tooltip="Project exists locally"
            disabled={true}
          >
            Imported
          </button>
          <button
            className="RemoteLabbooks__icon RemoteLabbooks__icon--delete"
            data-tooltip=""
            disabled={false}
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
        <div
          className="RemoteLabbooks__row RemoteLabbooks__row--text"
        >
          <div
            className="RemoteLabbooks__row RemoteLabbooks__row--title"
          >
            <h6
              className="RemoteLabbooks__panel-title"
            >
              <span>
                <span
                  className=""
                >
                  ui-test-remote
                </span>
              </span>
            </h6>
          </div>
          <p
            className="RemoteLabbooks__paragraph RemoteLabbooks__paragraph--owner"
          >
            uitest
          </p>
          <p
            className="RemoteLabbooks__paragraph RemoteLabbooks__paragraph--owner"
          >
            Created on Fri Jan 18 2019
          </p>
          <p
            className="RemoteLabbooks__paragraph RemoteLabbooks__paragraph--owner"
          >
            Modified Fri Jan 18 2019
          </p>
          <p
            className="RemoteLabbooks__paragraph RemoteLabbooks__paragraph--description"
          >
            <span>
              <span
                className=""
              >
                No Description
              </span>
            </span>
          </p>
        </div>
        <div
          className="Tooltip-Listing RemoteLabbooks__private   Tooltip-data Tooltip-data--small"
          data-tooltip="private"
        />
      </div>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
